qrenderdoc/Windows/PerformanceCounterViewer.cpp

```diff
@@ -39,17 +81,66 @@ PerformanceCounterViewer::PerformanceCounterViewer(ICaptureContext &ctx, QWidget
     {
       const QList<uint32_t> selectedCounters = pcs.GetSelectedCounters();
 
-      m_Ctx.Replay().AsyncInvoke([=](IReplayController *controller) -> void {
+      bool done = false;
+      m_Ctx.Replay().AsyncInvoke([=, &done](IReplayController *controller) -> void {
         rdctype::array<GPUCounter> counters;
         counters.create(selectedCounters.size());
 
+        QMap<GPUCounter, CounterDescription> counterDescriptions;
+
         for(int i = 0; i < selectedCounters.size(); ++i)
         {
           counters[i] = (GPUCounter)selectedCounters[i];
+          counterDescriptions.insert(counters[i], controller->DescribeCounter(counters[i]));
+        }
+
+        QMap<GPUCounter, int> counterIndex;
+        for(int i = 0; i < selectedCounters.size(); ++i)
+        {
+          counterIndex.insert((GPUCounter)selectedCounters[i], i);
         }
 
-        controller->FetchCounters(counters);
+        const auto results = controller->FetchCounters(counters);
+
+        GUIInvoke::call([this, results, counterDescriptions, counterIndex]() -> void {
+          ui->counterResults->clear();
+
+          QStringList headers;
+          headers << QLatin1String{"EID"};
+          for(const auto &cd : counterDescriptions)
+          {
+            headers << ToQStr(cd.name);
+          }
+
+          QMap<uint32_t, int> eventIdToRow;
+          for(const auto &result : results)
+          {
+            if(eventIdToRow.contains(result.eventID))
+              continue;
+            eventIdToRow[result.eventID] = eventIdToRow.size();
+          }
+
+          ui->counterResults->setColumnCount(headers.size());
+          ui->counterResults->setHorizontalHeaderLabels(headers);
+          ui->counterResults->setRowCount(eventIdToRow.size());
+          ui->counterResults->verticalHeader()->hide();
```

Comment:
If the vertical header is to be unconditionally hidden, prefer to do that in the `.ui` file so that it doesn't appear at first then disappear when results load.